mvg/utils/response_processing.py

```diff
@@ -55,3 +55,54 @@ def get_paginated_items(request: Callable, url: str, params: Dict) -> Dict:
             response = request("get", url, params=params)
             all_items += response.json()["items"]
     return {"items": all_items, "total": num_items}
+
+
+def get_paginated_analysis_results(request: Callable, url: str, params: Dict) -> Dict:
+    """
+    Iteratively construct the results of an analysis making use of the
+    pagination parameters provided in the params dictionary
+
+    Parameters
+    ----------
+    request : Callable
+        the request object
+    url : str
+        the URL to retrieve the analysis results
+    params : Dict
+        parameters dict for the request object
+
+    Returns
+    -------
+    Dict
+        Analysis results
+    """
+    # Pagination Model fields
+    paginator_model_fields = ["total", "limit", "offset"]
+    # Fields without paginated data
+    non_paginated_fields = ["mode_info", "mode_probabilities"]
```

Comment:
I think it is more understandable for the reader to know which fields are paginated rather than which ones are excluded. In the vibium-cloud we have it in a list

https://github.com/vikinganalytics/vibium-cloud/blob/e3063c4f4579042183a01afa24c24c3ef8196086/vibium_app/features/modeid.py#L292

This also simplifies the check in line 102 to
```python
                    if key in paginatable_fields:
                        results[key] += _results[key]
```